### PR TITLE
[Php74] Skip mixed type on TypedPropertyRector on auto import enabled

### LINKED ISSUE #6991: Error:  Mixed can only be used as a standalone type
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | 0.12.16             |

I ran rector on a project and according to the console output the rule named **_TypedPropertyRector_** was responsible for changing:

```php
/** @var Foo|mixed|MockObject */
private $x;
```

To:

```php
private mixed|\PHPUnit\Framework\MockObject\MockObject|\App\Services\Foo $x = null;
```

This change creates a fatal error: **Mixed can only be used as a standalone type**

## Minimal PHP Code Causing Issue

See https://getrector.org/demo/1ec8a2b5-0221-6fc0-ae53-e3ff0cc295bf

## Expected Behaviour

I can stop this from happening by editing the `rules/Php74/TypeAnalyzer/ObjectTypeAnalyzer.php isSpecial` function:

```php
foreach ($types as $type) {
    if ($type instanceof \PHPStan\Type\MixedType && count($types) > 1) {
        return true;
    }
```

But, a better solution would be to convert it to:

```php
/** @var Foo|MockObject */
private mixed $x;
```

### PR DESCRIPTION
**Pull request by dac514**
# Failing Test for TypedPropertyRector

Based on https://getrector.org/demo/1ec8a2b5-0221-6fc0-ae53-e3ff0cc295bf

Fixes https://github.com/rectorphp/rector/issues/6991

### CODE REVIEW

**Comment by samsonasik** [comment]
@connerbw could you create a patch for it as well? thank you.

File: rules-tests/Php74/Rector/Property/TypedPropertyRector/Fixture/demo_file.php.inc [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+<?php
+
+namespace Rector\Tests\Php74\Rector\Property\TypedPropertyRector\Fixture;
+
+use PHPUnit\Framework\MockObject\MockObject;
+
+class DemoFile
+{
+  /** @var \Foo|\PHPUnit\Framework\MockObject\MockObject */
+  private mixed $x;
```
Comment (samsonasik) [inline rules-tests/Php74/Rector/Property/TypedPropertyRector/Fixture/demo_file.php.inc]: mixed is not exists yet in php 7.4, so better skip it, ref https://3v4l.org/v1Vcd#v7.4.27

**Comment by dac514** [comment]
Oh, a patch would take me longer.
In https://github.com/rectorphp/rector/issues/6991 I only know how to **_skip_** the problem.
This testcase is for the ideal scenario. I'll have think about it.

File: rules-tests/Php74/Rector/Property/TypedPropertyRector/Fixture/demo_file.php.inc (hunk 1 above)
Comment (dac514) [reply rules-tests/Php74/Rector/Property/TypedPropertyRector/Fixture/demo_file.php.inc]: Ok. So we do want to skip. I'll amend the PR.

File: rules-tests/Php74/Rector/Property/TypedPropertyRector/Fixture/demo_file.php.inc [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+namespace Rector\Tests\Php74\Rector\Property\TypedPropertyRector\Fixture;
+
+use PHPUnit\Framework\MockObject\MockObject;
+
+class DemoFile
+{
+  /** @var Foo|mixed|MockObject */
+  private $x;
+}
+?>
```
Comment (samsonasik) [inline rules-tests/Php74/Rector/Property/TypedPropertyRector/Fixture/demo_file.php.inc]: This can be removed

```suggestion
```

**Comment by samsonasik** [comment]
Please rename fixture to:

- skip_mixed_type.php.inc
- with class SkipMixedType

**Comment by samsonasik** [comment]
CI seems green, do you think it is still issue on rector 0.12.16, or it is only happen on auto import enabled? If yes, you may need to move to auto import directory https://github.com/rectorphp/rector-src/tree/main/rules-tests/Php74/Rector/Property/TypedPropertyRector/FixtureImported

**Comment by dac514** [comment]
I can reproduce here:  https://getrector.org/demo/1ec8a2b5-0221-6fc0-ae53-e3ff0cc295bf

**Comment by samsonasik** [comment]
getrector.org still using rector 0.12.15

**Comment by ls-dac-chartrand** [comment]
> getrector.org still using rector 0.12.15

Oh? Ok.

**Comment by samsonasik** [comment]
Thank you @ls-dac-chartrand @connerbw

**Comment by samsonasik** [comment]
mixed type seems need to be possible if the type is not union type, and in php 8.0, so need to be checked early, and add new fixture to `rules-tests/Php74/Rector/Property/TypedPropertyRector/FixtureUnionIntersectionTypes`, 

ref https://3v4l.org/Dup0g

**Pull request merged**